R/Subsets.R

```diff
@@ -402,17 +427,28 @@ CohortSubsetOperator <- R6::R6Class(
 #'
 #' @param startWindow               A SubsetCohortWindow that patients must fall inside (see createSubsetCohortWindow)
 #' @param endWindow                 A SubsetCohortWindow that patients must fall inside (see createSubsetCohortWindow)
+#' @param windows                   A list of additional windows to be applied (logic is to always apply these with AND conditions)
 #' @param negate                    The opposite of this definition - include patients who do NOT meet the specified criteria
 #' @returns a CohortSubsetOperator instance
-createCohortSubset <- function(name = NULL, cohortIds, cohortCombinationOperator, negate, startWindow, endWindow) {
+createCohortSubset <- function(name = NULL, cohortIds, cohortCombinationOperator, negate, windows = list(), startWindow = NULL, endWindow = NULL) {
   subset <- CohortSubsetOperator$new()
   subset$name <- name
   subset$cohortIds <- cohortIds
   subset$cohortCombinationOperator <- cohortCombinationOperator
   subset$negate <- negate
-  subset$startWindow <- startWindow
-  subset$endWindow <- endWindow
 
+  # Start and end windows must always have subset anchor values set to support backwards compatibility
+  if (!is.null(startWindow)){
```

Comment:
THis section forces the old logic upon the user, which I dislike, but it also preseves the exact logic of the previous implementation. To me there is no perfect solution here...